src/WebJobs.Script.WebHost/Diagnostics/SystemLogger.cs

```diff
@@ -37,6 +47,20 @@ public class SystemLogger : ILogger
             _debugStateProvider = debugStateProvider;
             _eventManager = eventManager;
             _scopeProvider = scopeProvider;
+            _logItemQueue = new List<LogItem>();
```

Comment:
I've moved the actual emitting of the logs to a batched background task. This allows expensive log operations (e.g. the SystemTraceMiddleware Executing/Executed logs which do json serialization of request data) to the background, by having the call site pass a log formatter function so the operation is delayed, off the hot path.